workflow/scripts/ts_processors.py

```diff
@@ -198,110 +198,63 @@ class tskit_windowed_sfs_plus_ld(BaseProcessor):
-        Gap sizes follow power 2 distribution.
-        The LD is then computed and averaged over different bin (distance_bins) sizes.
-
-        Parameters ---------- haplotype : numpy 2D array or allel.haplotype SNP matrix where in the first dimension are
-        the SNP (rows) and in the second dimension (columns) are the samples. pos_vec : 1D array array of absolute
-        positions in [0, size_chr]. size_chr : int Size of the chromosome. circular : bool Whether to consider the
-        chromosome circular or not. If circular, the maximum distance between 2 SNPs is thus half the chromosome.
-        distance_bins : int or list LD will be averaged by bins of distances e.g. if distance_bins = [0, 100, 1000,
-        10000], LD will be averaged for the groups [0,100[, [100, 1000[, and [1000, 10000[ If distance_bins is an int,
-        it defines the number of bins of distances for which to compute the LD The bins are created in a logspace If
-        distance_bins is a list, they will be used instead
-
-        Returns
-        -------
-        DataFrame
-            Table with the distance_bins as index, and the mean value of
+        Compute LD for a subset of SNPs and return a DataFrame with only the mean r2.
         """
+        # Set up distance bins if not provided (kept here for potential grouping)
         if distance_bins is None or isinstance(distance_bins, int):
             if isinstance(distance_bins, int):
                 n_bins = distance_bins - 1
             else:
                 n_bins = 19
-            if circular:
-                distance_bins = np.logspace(2, np.log10(size_chr // 2), n_bins)
-                distance_bins = np.insert(distance_bins, 0, [0])
-            else:
-                distance_bins = np.logspace(2, np.log10(size_chr), n_bins)
-                distance_bins = np.insert(distance_bins, 0, [0])
+            distance_bins = np.logspace(2, np.log10(size_chr), n_bins)
+            distance_bins = np.insert(distance_bins, 0, [0])
 
-        # Iterate through gap sizes
         n_SNP, n_samples = haplotype.shape
         gaps = (2 ** np.arange(0, np.log2(n_SNP), 1)).astype(int)
 
-        # Initialize lists to store selected SNP pairs and LD values
         selected_snps = []
         for gap in gaps:
             snps = np.arange(0, n_SNP, gap) + np.random.randint(0, (n_SNP - 1) % gap + 1)
-            # adding a random start (+1, bc 2nd bound in randint is exlusive)
-
-            # non overlapping contiguous pairs
-            # snps=[ 196, 1220, 2244] becomes
-            # snp_pairs=[(196, 1220), (1221, 2245)]
-            snp_pairs = np.unique([((snps[i] + i) % n_SNP, (snps[i + 1] + i) % n_SNP) for i in range(len(snps) - 1)],
-                                  axis=0)
+            snp_pairs = np.unique([((snps[i] + i) % n_SNP, (snps[i + 1] + i) % n_SNP)
+                                for i in range(len(snps) - 1)], axis=0)
```

Comment:
for my eyes, it's way easier to have the list comprehension on a single line, splitting list comprehensions across lines makes things harder to read